app/assets/javascripts/discourse/app/components/topic-timeline/container.js

```diff
@@ -269,13 +296,51 @@ export default class TopicTimelineScrollArea extends Component {
     this.current = e.postIndex;
     this.percentage = e.percent;
     this.calculatePosition();
+    this.dockCheck();
   }
 
   @action
   goBack() {
     this.args.jumpToIndex(this.lastRead);
   }
 
+  dockCheck() {
+    const timeline = document.querySelector(".timeline-container");
+    const timelineHeight = (timeline && timeline.offsetHeight) || 400;
+
+    const prevDockAt = this.dockAt;
+    const positionTop = headerOffset() + window.pageYOffset;
+    const currentPosition = positionTop + timelineHeight;
+
+    this.dockBottom = false;
+    if (positionTop < this.topicTop) {
+      this.dockAt = parseInt(this.topicTop, 10);
+    } else if (currentPosition > this.topicBottom) {
+      this.dockAt = parseInt(this.topicBottom - timelineHeight, 10);
+      this.dockBottom = true;
+      if (this.dockAt < 0) {
+        this.dockAt = 0;
+      }
+    } else {
+      this.dockAt = null;
+    }
+
+    if (this.dockAt !== prevDockAt) {
+      const timelineContainer = document.querySelector(".timeline-container");
```

Comment:
Accessing and modifying elements from the parent component feels kinda messy. If we want to have `dockCheck()` here then I think we need to send class changes up to the parent using an action. So when it's docked, it would become something like

```
this.args.setDocked(true)
```

Reply:
Or maybe all of this, including the `postScrolled` hook should live in the parent component? I don't really mind either way, but I do think we should be aiming to avoid using `document.querySelector` to modify other components.

Reply:
Updated via [ead6d94](https://github.com/discourse/discourse/pull/20166/commits/ead6d94715486bf8407991e8d7e95492246d9013)